src/models/IssueTableStorage.gs

```diff
@@ -0,0 +1,127 @@
+
+  // sanitize and concat a new id string which is safe to use as an property key name
+  var tableIndexId = this.tableIndexName(IssueTable.data.sheetId, IssueTable.data.tableId);
+  this.tables_[tableIndexId] = IssueTable.toJson(); // add IssueTables data to index
+  debug.log('table with id %s added to index of tables', tableIndexId);
+
+  this.save();
+
+  return this;
+};
+
+/**
+ * Return a perviously stored IssueTable if available.
+ * 
+ * @param {string} tableId    Table Id to fetch
+ * @param {sheetId} sheetId    Optional: Sheet Id to fetch table for. Default: current active sheets id
+ * @return {IssueTable_}    Returns instance of IssueTable with loaded data if found, or empty instance if not
+ */
+IssueTableIndex_.prototype.getTable = function(tableId, sheetId) {
+  sheetId = sheetIdPropertySave(sheetId);
+  debug.log('getTable(%s, %s)', tableId, sheetId);
+
+  // load everything from storage
+  this.load_();
+
+  //sanitize and concat a new id string which is safe to use as an property key name
+  var tableIndexId = this.tableIndexName(sheetId, tableId);
+  // fetch table from storage by tableIndexId
+  this.tables_[tableIndexId] = this.storage_.getValue(tableIndexId);
+
+  if (!this.tables_.hasOwnProperty(tableIndexId)) {
+    debug.log('No table with indexId %s found.', tableIndexId);
+    return false;
+  }
+
+  // Initialize new IssueTable object and load data from storage into it
+  var IssueTable = new IssueTable_();
+  return IssueTable.fromJson(this.tables_[tableIndexId]);
+};
+
+/**
+ * Loading all available relevant data from storage
+ * @return {IssueTableIndex_}    Allow chaining
+ */
+IssueTableIndex_.prototype.load_ = function() {
+  debug.log('IssueTableIndex_.load_()');
+  var index = this.storage_.getValue('index');
+
+  if (null !== index) {
+    this.index_ = index;
+  }
+
+  for ( var sheetId in this.index_) {
+    for ( var tableId in this.index_[sheetId]) {
+      var tableIndexId = this.tableIndexName(sheetId, tableId);
+      this.tables_[tableIndexId] = this.storage_.getValue(tableIndexId);
+    }
+  }
+
+  return this;
+};
+
+/**
+ * Save all data to storage
+ * @return {IssueTableIndex_}    Allow chaining
+ */
+IssueTableIndex_.prototype.save = function() {
+  debug.log('IssueTableIndex_.save()');
+  // store the index of all tables
+  this.storage_.setValue('index', this.index_);
+
+  // store each table individually into the property storage
+  for ( var element in this.tables_) {
+    this.storage_.setValue(element, this.tables_[element]);
+  }
+
+  return this;
+};
+
+/**
+ * Format a usable string from sheetId and tableId
+ * @param {string} sheetId    The sheet's sanitized id
+ * @param {string} tableId    The table id
+ * @return {string}
+ */
+IssueTableIndex_.prototype.tableIndexName = function(sheetId, tableId) {
+  return sheetId + '__' + tableId;
+};
+
+// Node required code block
+module.exports = {
+  IssueTableStorage_ : IssueTableStorage_
+};
+// End of Node required code block
```

Comment:
There are a few patterns for classes in the codebase (and in this PR) - do you have a preference - I don't but maybe it should use the same one throughout.

Reply:
Agree, what different pattern do you found or which one you suggest?

Reply:
Whichever pattern is mostly used already :) Less refactoring. The one you used in Search.gs and IssueTable

Reply:
True yeah... im not sure actually. Latest code is using the pattern i saw on most of the "official" google code...  again, im usually not developing JS that much :)

I can see benefits for both...  if you prefer too, then lets stick to the old one, i used in `search.gs` and `IssueTable.gs`

Reply:
The official way is to write code in ES6 and use the new class constructs. Everything before then is a bit of a hack
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes

Gas isnt compatible with ES6 but we could use a transpiler like babel during the gulp build to convert the code before upload.
This project does that:-
https://github.com/labnol/apps-script-starter
That's a way off for this codebase I think and will make it hard to edit the transpiled code on the fly in the GAS environment

So for now we just have to pick one of the hacky ways to do it. I'm good with the one used in search.gs (thats one of the nicest pieces of code in the code base in terms of readablity)

Reply:
Lets do it then like `search.gs`!